src/modules/navigator/mission.cpp

```diff
@@ -1912,6 +1912,7 @@ bool Mission::position_setpoint_equal(const position_setpoint_s *p1, const posit
 		(p1->acceleration_is_force == p2->acceleration_is_force) &&
 		(fabsf(p1->acceptance_radius - p2->acceptance_radius) < FLT_EPSILON) &&
 		(fabsf(p1->cruising_speed - p2->cruising_speed) < FLT_EPSILON) &&
-		(fabsf(p1->cruising_throttle - p2->cruising_throttle) < FLT_EPSILON));
+		((fabsf(p1->cruising_throttle - p2->cruising_throttle) < FLT_EPSILON) || (!PX4_ISFINITE(p1->cruising_throttle)
+				&& !PX4_ISFINITE(p2->cruising_throttle)));
```

Comment:
@dagar I couldn't think of any better way to handle the nans in this context

Reply:
Yes I think that makes sense, although maintaining this entire thing is pretty gross. We should look at generating comparison operators for uORB later. https://github.com/PX4/Firmware/blob/df38e3798c25541d1462d8011fbf0d58904ce4f7/msg/templates/uorb/msg.cpp.em#L85-L99

Reply:
@dagar Right, makes sense